### PR TITLE
mgr/dashboard: remove password field if login is using SSO and fix error message in confirm password

### PR DESCRIPTION
**Pull request by ishanrai05**
Signed-off-by: Ishan Rai <ishanrai05@gmail.com>

fixes: https://tracker.ceph.com/issues/46494

## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by ishanrai05** [comment]
#### Before:
It is possible to create user without entering a password
Screenshot from 2020-07-28 14-15-47

#### After:
Screenshot from 2020-07-28 14-16-25

**Comment by votdev** [comment]
This PR will deny users to use empty passwords. That's not a good idea from a security perspective, but we shouldn't prevent users doing that.

**Comment by votdev** [comment]
There is a password policy which allows the admin to set the minimum length of the password. If this is enabled, the form submission will be blocked until the length validation succeeds. If the password validation fails, the input field will be marked invalid/red. Adding the `required` flag to the input field will block the ability to use empty passwords. IMO we should not do that because we really don't know if there is a user out there that WANTS to do that because of a special reason. We already have options to allow the user to configure individual password policies.

**Comment by Devp00l** [comment]
If the admin has not set a password a user can't log in anyway, so why change it?

**Comment by tspmelo** [comment]
Since, as mention before, there might be some situation where the admin wants to create users without password (which won't be able to login). I would recommend we keep it as it is and close that issue.

I would propose you create a new PR, or modify this one, with the following:
- Keep the fix you did with the error messages for the `confirmpassword`, guessing there was some visual problem.
- Remove both password fields, if the login is using SSO. you can check this with `AuthStorageService.isSSO()`

**Comment by ishanrai05** [comment]
#### While using SSO

SSO

**Comment by ishanrai05** [comment]
#### Using SSO

sso2

**Comment by tspmelo** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html [hunk 1]
```diff
@@ -93,13 +95,13 @@
                         cdPasswordButton="confirmpassword">
                 </button>
               </span>
+              <span class="invalid-feedback"
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.html:98]: Any reason why this has been relocated?

**Comment by avanthakkar** [comment]
Can you please add the tracker link for this issue to the PR description and in commit message too?

**Comment by ishanrai05** [comment]
@avanthakkar There is no tracker issue for this PR. Should I open one?

**Comment by LenzGr** [comment]
> Should I open one?

Yes, please! This way, there is a reference that provides some background information about why this change is needed (and makes it possible to earmark it for backporting, if necessary).

**Comment by tspmelo** [comment]
Didn't we had a tracker issue before? Since we decided to change how to fix the previous issue I think we could still use it and just add more information to the issue.

**Comment by avanthakkar** [comment]
> Didn't we had a tracker issue before? Since we decided to change how to fix the previous issue I think we could still use it and just add more information to the issue.

Yes, I think this is the one https://tracker.ceph.com/issues/46494, right?

**Comment by avanthakkar** [comment]
jenkins test api

**Pull request merged**